Add publora extension

T0ha opened the pull request:
## Description

Helps to managed and schedule social media posts through [Publora](https://app.publora.com). 
Currently, supports only scheduling new post due to API limitations.
Going to add searching and management existing stuff. 

## Screencast

![Screenshot](https://github.com/T0ha/publora-raycast/blob/45bf5c0ee976d4b07f16e64f7198d3ff638106a1/metadata/publora-1.png?raw=true)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

T0ha commented on extensions/publora/package.json:
```diff
@@ -0,0 +1,47 @@
+    {
+      "name": "schedule-post",
+      "title": "Schedule Post",
+      "description": "Create and shedule a post in Publora",
```
Fixed

T0ha commented on extensions/publora/src/schedule-post.tsx:
```diff
@@ -0,0 +1,84 @@
+    setContent("");
+    setDue(null);
+    setPlatforms([]);
+  }
```
Fixed

T0ha commented:
@geptilebot, I've fixed issues, pls, review

LitoMore commented on extensions/publora/src/api.tsx:
```diff
@@ -0,0 +1,37 @@
+    throw new Error(`Error scheduling post: ${errorData.message}`);
+  }
+
+  return await response.json();
```
Be careful when writing `return await`. See here: https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise.

```suggestion
  return response.json();
```

T0ha replied:
Thank you 🙏 I'm mostly Elixir dev for now. I have no too much experience in JS / TS. I'll educate myself.

T0ha replied:
Fixed

LitoMore commented on extensions/publora/src/schedule-post.tsx:
```diff
@@ -0,0 +1,89 @@
+    }
+  }
+
+  const api_key = getPreferenceValues<{ api_key: string }>().api_key;
```
There is an `ExtensionPreferences` in the global scope. You can simply pass in the `ExtensionPreferences` its generic type.

```suggestion
  const api_key = getPreferenceValues<ExtensionPreferences>().api_key;
```

T0ha replied:
Fixed

T0ha commented:
@pernielsentikaer @LitoMore I've refactored to `useForm` and added validation and `API  key error`. But I couldn't find a way to clear form after successful schedule. Could you, pls, suggest the way?

pernielsentikaer commented:
I updated the code, can you give it a test to see if it's working for you?
